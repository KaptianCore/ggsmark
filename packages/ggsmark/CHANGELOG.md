# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.8.3](https://github.com/johnnyhuy/ggsmark/compare/ggsmark@0.8.2...ggsmark@0.8.3) (2020-09-13)


### Bug Fixes

* move index out to main folder ([ee60135](https://github.com/johnnyhuy/ggsmark/commit/ee60135b993d26a7c139046ea5f9c36f1e5f8b9c))





## [0.8.2](https://github.com/johnnyhuy/ggsmark/compare/ggsmark@0.8.1...ggsmark@0.8.2) (2020-09-13)


### Bug Fixes

* Revert "fix: try dist" ([8d6d36e](https://github.com/johnnyhuy/ggsmark/commit/8d6d36e60408532a9218b5db0597540b8bf33bdb))
* Revert "fix: Use dist" ([f838486](https://github.com/johnnyhuy/ggsmark/commit/f8384861d7b043439a67dc903a9780dfafa8267e))
* try dist ([d02b34c](https://github.com/johnnyhuy/ggsmark/commit/d02b34c43b9d02ae0ee156113426954e1d0b5bc3))
* try src folder again ([5329336](https://github.com/johnnyhuy/ggsmark/commit/5329336cbb7456bc9fd844fcc43bbaa6c1fcc169))
* Use dist ([e030308](https://github.com/johnnyhuy/ggsmark/commit/e03030845ada2600a25b0bd894844eedb4fc1358))





## [0.8.1](https://github.com/johnnyhuy/ggsmark/compare/ggsmark@0.8.0...ggsmark@0.8.1) (2020-09-13)

**Note:** Version bump only for package ggsmark





# [0.8.0](https://github.com/johnnyhuy/ggsmark/compare/ggsmark@0.7.0...ggsmark@0.8.0) (2020-09-13)


### Features

* added core JS ([1d9e424](https://github.com/johnnyhuy/ggsmark/commit/1d9e424426cdf2ddae38c817e3d3f23d436cfca4))
* put packages in src folder ([d10a8bb](https://github.com/johnnyhuy/ggsmark/commit/d10a8bb233cf6140a0d0f37b2f8ae2670eeefe2a))





# [0.7.0](https://github.com/johnnyhuy/ggsmark/compare/ggsmark@0.6.0...ggsmark@0.7.0) (2020-09-13)


### Features

* Added browser and module files ([9581389](https://github.com/johnnyhuy/ggsmark/commit/958138995ac17ed1eb3a9b13d2abb8b2e42521d5))





# [0.6.0](https://github.com/johnnyhuy/ggsmark/compare/ggsmark@0.5.1...ggsmark@0.6.0) (2020-09-13)


### Bug Fixes

* don't ignore tests in main Babel ([bf2ecd2](https://github.com/johnnyhuy/ggsmark/commit/bf2ecd21c81d11522b13b3631bfd1ef877244271))


### Features

* Export to index JS ([08a5dc0](https://github.com/johnnyhuy/ggsmark/commit/08a5dc0e5f4278bf7fbccf23eb99647c5a82dbc3))





## [0.5.1](https://github.com/johnnyhuy/ggsmark/compare/ggsmark@0.5.0...ggsmark@0.5.1) (2020-09-13)

**Note:** Version bump only for package ggsmark





# [0.5.0](https://github.com/johnnyhuy/ggsmark/compare/ggsmark@0.4.1...ggsmark@0.5.0) (2020-09-13)


### Features

* added spoilers ([ec91cf1](https://github.com/johnnyhuy/ggsmark/commit/ec91cf11cc1f9e00eefd77310bb64d4fd28fefa5))





## [0.4.1](https://github.com/johnnyhuy/ggsmark/compare/ggsmark@0.4.0...ggsmark@0.4.1) (2020-09-13)

**Note:** Version bump only for package ggsmark
