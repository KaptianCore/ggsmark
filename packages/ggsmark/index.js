import unified from 'unified'
import markdown from 'remark-parse'
import merge from 'deepmerge'
import gh from 'hast-util-sanitize/lib/github'
import sanitize from 'rehype-sanitize'

// Plugins
import iframe from 'remark-iframes'
import align from 'remark-text-alignment'
import color from 'remark-color-text'

// Don't use remark-html otherwise we can't customize HTML
import stringify from 'rehype-stringify'
import rehype from 'remark-rehype'

// Import this since remark-iframe needs it
import 'regenerator-runtime/runtime'

export default (text, options = {}) => {
  let schema = merge(gh, {
    attributes: {
      '*': [
        'allowfullscreen',
        'frameborder',
        'src',
        'className',
        'style',
        'width',
        'height'
      ]
    },
    tagNames: ['iframe']
  })
  return unified()
    .use(markdown, {
      blocks: []
    })
    .use(iframe, {
      'www.youtube.com': {
        tag: 'iframe',
        width: 560,
        height: 315,
        disabled: false,
        replace: [
          ['watch?v=', 'embed/'],
          ['http://', 'https://']
        ],
        thumbnail: {
          format: 'http://img.youtube.com/vi/{id}/0.jpg',
          id: '.+/(.+)$'
        },
        removeAfter: '&'
      },
      'youtu.be': {
        tag: 'iframe',
        width: 560,
        height: 315,
        disabled: false,
        replace: [
          ['watch?v=', 'embed/'],
          ['http://', 'https://']
        ],
        thumbnail: {
          format: 'http://img.youtube.com/vi/{id}/0.jpg',
          id: '.+/(.+)$'
        },
        removeAfter: '&'
      },
      'soundcloud.com': {
        tag: 'iframe',
        width: '100%',
        height: 150,
        disabled: false,
        replace: [
          [
            'soundcloud.com/',
            'w.soundcloud.com/player/?visual=true&url=https://soundcloud.com/'
          ],
          ['http://', 'https://']
        ]
      },
      'www.clips.twitch.tv': {
        tag: 'iframe',
        width: '100%',
        height: 378,
        disabled: false,
        replace: [
          [
            'www.clips.twitch.tv/',
            `www.clips.twitch.tv/embed?parent=${options.parent}&clip=`
          ],
          ['http://', 'https://']
        ]
      },
      'clips.twitch.tv': {
        tag: 'iframe',
        width: '100%',
        height: 378,
        disabled: false,
        replace: [
          [
            'clips.twitch.tv',
            `clips.twitch.tv/embed?parent=${options.parent}&clip=`
          ],
          ['http://', 'https://']
        ]
      },
      'www.twitch.tv': {
        tag: 'iframe',
        width: 620,
        height: 378,
        disabled: false,
        replace: [
          [
            'www.twitch.tv/videos/',
            `www.player.twitch.tv/?${options.parents}&video=`
          ],
          ['http://', 'https://']
        ]
      },
      'twitch.tv': {
        tag: 'iframe',
        width: 620,
        height: 378,
        disabled: false,
        replace: [
          [
            'twitch.tv/videos/',
            'player.twitch.tv/?parent=${options.parent}&video='
          ],
          ['http://', 'https://']
        ]
      }
    })
    .use(align)
    .use(rehype)
    .use(stringify)
    .use(color)
    .use(sanitize, schema)
    .processSync(text)
    .toString()
}
