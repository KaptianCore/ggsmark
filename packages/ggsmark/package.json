{
  "name": "ggsmark",
  "version": "0.4.0",
  "description": "Markdown extension for the Gentlemen's Gaming Society website.",
  "repository": "https://github.com/johnnyhuy/ggsmark.git",
  "author": "Johnny Huynh <info@johnnyhuy.com>",
  "license": "MIT",
  "dependencies": {
    "rehype-stringify": "^8.0.0",
    "remark-iframes": "^4.0.4",
    "remark-parse": "^8.0.3",
    "remark-rehype": "^7.0.0",
    "remark-stringify": "^8.1.1",
    "remark-color-text": "^0.0.1",
    "remark-text-alignment": "^0.2.1",
    "unified": "^9.2.0"
  }
}
