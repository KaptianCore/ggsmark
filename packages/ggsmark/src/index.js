import unified from 'unified'
import markdown from 'remark-parse'
import merge from 'deepmerge'
import gh from 'hast-util-sanitize/lib/github'
import sanitize from 'rehype-sanitize'

// Plugins
import iframe from 'remark-iframes'
import align from 'remark-text-alignment'
import color from 'remark-color-text'
import spoilers from 'remark-spoilers'

// Don't use remark-html otherwise we can't customize HTML
import stringify from 'rehype-stringify'
import rehype from 'remark-rehype'

// Import this since remark-iframe needs it
import 'regenerator-runtime/runtime'

export default (text, options = {}) => {
  options.parents =
    '?' + options.parents.map((i) => 'parent=' + i).join('&') ?? []

  let schema = merge(gh, {
    attributes: {
      '*': [
        'allowfullscreen',
        'frameborder',
        'src',
        'className',
        'style',
        'width',
        'height'
      ]
    },
    tagNames: ['iframe']
  })

  return unified()
    .use(markdown, {
      blocks: []
    })
    .use(iframe, {
      'www.youtube.com': {
        tag: 'iframe',
        width: 560,
        height: 315,
        disabled: false,
        replace: [
          ['watch?v=', 'embed/'],
          ['http://', 'https://']
        ],
        thumbnail: {
          format: 'http://img.youtube.com/vi/{id}/0.jpg',
          id: '.+/(.+)$'
        },
        removeAfter: '&'
      },
      'youtu.be': {
        tag: 'iframe',
        width: 560,
        height: 315,
        disabled: false,
        replace: [
          ['watch?v=', 'embed/'],
          ['http://', 'https://']
        ],
        thumbnail: {
          format: 'http://img.youtube.com/vi/{id}/0.jpg',
          id: '.+/(.+)$'
        },
        removeAfter: '&'
      },
      'soundcloud.com': {
        tag: 'iframe',
        width: '100%',
        height: 150,
        disabled: false,
        replace: [
          [
            'soundcloud.com/',
            'w.soundcloud.com/player/?visual=true&url=https://soundcloud.com/'
          ],
          ['http://', 'https://']
        ]
      },
      'www.clips.twitch.tv': {
        tag: 'iframe',
        width: 620,
        height: 378,
        disabled: false,
        replace: [
          [
            'https://clips.twitch.tv/',
            'https://clips.twitch.tv/embed?parent=ggs.sx&clip='
          ],
          ['http://', 'https://']
        ]
      },
      'clips.twitch.tv': {
        tag: 'iframe',
        width: 620,
        height: 378,
        disabled: false,
        replace: [
          [
            'https://clips.twitch.tv',
            'https://clips.twitch.tv/embed?parent=ggs.sx&clip='
          ],
          ['http://', 'https://']
        ]
      }
    })
    .use(align)
    .use(rehype)
    .use(stringify)
    .use(color)
    .use(spoilers)
    .use(sanitize, schema)
    .processSync(text)
    .toString()
}
