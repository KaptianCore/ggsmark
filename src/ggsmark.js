import unified from 'unified'
import markdown from 'remark-parse'
import stringify from 'rehype-stringify'
import rehype from 'remark-rehype'
import iframe from 'remark-iframes'
import remarkAlign from 'remark-align'

// Import this since remark-iframe needs it
import 'regenerator-runtime/runtime'

export default (text) => {
  return unified()
    .use(markdown, {
      blocks: []
    })
    .use(iframe, {
      'www.youtube.com': {
        tag: 'iframe',
        width: 560,
        height: 315,
        disabled: false,
        replace: [
          ['watch?v=', 'embed/'],
          ['http://', 'https://']
        ],
        thumbnail: {
          format: 'http://img.youtube.com/vi/{id}/0.jpg',
          id: '.+/(.+)$'
        },
        removeAfter: '&'
      },
      'youtu.be': {
        tag: 'iframe',
        width: 560,
        height: 315,
        disabled: false,
        replace: [
          ['watch?v=', 'embed/'],
          ['http://', 'https://']
        ],
        thumbnail: {
          format: 'http://img.youtube.com/vi/{id}/0.jpg',
          id: '.+/(.+)$'
        },
        removeAfter: '&'
      },
      'soundcloud.com': {
        tag: 'iframe',
        width: '100%',
        height: 150,
        disabled: false,
        replace: [
          [
            'soundcloud.com/',
            'w.soundcloud.com/player/?visual=true&url=https://soundcloud.com/'
          ],
          ['http://', 'https://']
        ]
      }
    })
    .use(markdown)
    .use(remarkAlign, {
      left: 'align-left',
      center: 'align-center',
      right: 'align-right',
    })
    .use(rehype)
    .use(stringify)
    .processSync(text)
    .toString()
}
