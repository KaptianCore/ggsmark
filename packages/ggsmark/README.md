# GGSMark

![npm](https://img.shields.io/npm/v/ggsmark)

Markdown package used in the [GGS.SX](https://ggs.sx/) website.

## Installation

```bash
npm i ggsmark
```

## Usage

```js
import ggsmark from 'ggsmark'

let output = ggsmark('**foo** bar')

console.log(output) // <p><strong>foo</strong> bar</p>
```

It's also worth investigating unit tests to understand the expected output.

## Examples

### Strikethrough

```markdown
~~Text~~
```

### SoundCloud

```markdown
!(https://soundcloud.com/djtechnoboy/tnt-sound-rush-right-now)
```

### YouTube

```markdown
!(https://www.youtube.com/watch?v=pwmY1XUTBpE)
```
