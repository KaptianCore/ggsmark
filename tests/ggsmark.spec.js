import ggsmark from '../src/ggsmark'
import axios from 'axios'

jest.mock('axios')

// describe('render text alignment', () => {
//   test('text center', () => {

//     // Arrange
//     let string = `:text-right\n# Heading\n\n**bold text**\ntest\n:text-right\ntest`

//     // Act
//     let result = ggsmark(string)

//     // Assert
//     expect(result).toMatchSnapshot()
//   })
// })

describe('render soundcloud blocks', () => {
  test('soundcloud iframe test', () => {
    // Arrange
    const response = {
      data: {
        version: 1.0,
        type: 'rich',
        provider_name: 'SoundCloud',
        provider_url: 'https://soundcloud.com',
        height: 166,
        width: '100%',
        title: 'WAP feat. Megan Thee Stallion by Cardi B',
        description:
          'Cardi B - WAP feat. Megan Thee Stallion\nStream/Download: https://CardiB.lnk.to/WAP\n\nSubscribe for more official content from Cardi B: https://CardiB.lnk.to/Subscribe\n\nFollow Cardi B\nhttp://cardibofficial.com\nhttp://Twitter.com/IAmCardiB\nhttp://Facebook.com/IAmCardiB\nhttp://Instagram.com/f/iamcardib\nhttp://Soundcloud.com/IAmCardiB\n\nExclusive Bardi Gang merchandise available here: http://smarturl.it/BardiGangMerchYT',
        thumbnail_url:
          'https://i1.sndcdn.com/artworks-xRcHRzwpJGpfGt1d-wqhSGQ-t500x500.jpg',
        html:
          '<iframe width="100%" height="166" scrolling="no" frameborder="no" src="https://w.soundcloud.com/player/?visual=true&url=https%3A%2F%2Fapi.soundcloud.com%2Ftracks%2F871426135&show_artwork=true&maxheight=166"></iframe>',
        author_name: 'Cardi B',
        author_url: 'https://soundcloud.com/iamcardib'
      }
    }
    axios.get.mockResolvedValue(response)
    let string = `:soundcloud https://soundcloud.com/iamcardib/wap-feat-megan-thee-stallion`

    // Act
    let result = ggsmark(string)

    // Assert
    expect(result)
      .toBe(`<iframe width=\"100%\" height=\"166\" scrolling=\"no\" frameborder=\"no\" src=\"https://w.soundcloud.com/player/?visual=true&url=https%3A%2F%2Fapi.soundcloud.com%2Ftracks%2F871426135&show_artwork=true&maxheight=166\"></iframe>
    `)
  })
})

describe('render youtube blocks', () => {
  test('repeated youtube with text before and after', () => {
    // Arrange
    let string = `
**bold** string before youtube :youtube http://www.youtube.com/watch?v=52c_QSg64fs after youtube :youtube http://www.youtube.com/watch?v=waefawefwaef *italics*
soft new line

new line
    `

    // Act
    let result = ggsmark(string)

    // Assert
    expect(result).toMatchSnapshot()
  })

  test('repeated youtube', () => {
    // Arrange
    let string = `:youtube http://www.youtube.com/watch?v=52c_QSg64fs :youtube http://www.youtube.com/watch?v=52c_QSg64fs :youtube http://www.youshit.com`

    // Act
    let result = ggsmark(string)

    // Assert
    expect(result).toMatchSnapshot()
  })

  test('youtube with timestamp', () => {
    // Arrange
    let string = `:youtube https://youtu.be/IJhgZBn-LHg?t=712`

    // Act
    let result = ggsmark(string)

    // Assert
    expect(result).toMatchSnapshot()
  })
})
