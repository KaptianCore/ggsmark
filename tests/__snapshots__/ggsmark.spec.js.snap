// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render youtube blocks repeated youtube 1`] = `
"<p>
</p><div class=\\"youtube-video\\">
<iframe type=\\"text/html\\" src=\\"https://www.youtube.com/embed/52c_QSg64fs\\"></iframe>
</div>
 
<div class=\\"youtube-video\\">
<iframe type=\\"text/html\\" src=\\"https://www.youtube.com/embed/52c_QSg64fs\\"></iframe>
</div>
 :youtube http://www.youshit.com<p></p>
"
`;

exports[`render youtube blocks repeated youtube with text before and after 1`] = `
"<p><strong>bold</strong> string before youtube 
</p><div class=\\"youtube-video\\">
<iframe type=\\"text/html\\" src=\\"https://www.youtube.com/embed/52c_QSg64fs\\"></iframe>
</div>
 after youtube 
<div class=\\"youtube-video\\">
<iframe type=\\"text/html\\" src=\\"https://www.youtube.com/embed/waefawefwaef\\"></iframe>
</div>
 <em>italics</em>
soft new line<p></p>
<p>new line</p>
"
`;

exports[`render youtube blocks youtube with timestamp 1`] = `
"<p>
</p><div class=\\"youtube-video\\">
<iframe type=\\"text/html\\" src=\\"https://www.youtube.com/embed/IJhgZBn-LHg?t=712\\"></iframe>
</div>
<p></p>
"
`;
